rust/xaynet-core/Cargo.toml

```diff
@@ -11,6 +11,10 @@ license-file = "../../LICENSE"
 keywords = ["federated-learning", "fl", "ai", "machine-learning"]
 categories = ["science", "cryptography"]
 
+[package.metadata.docs.rs]
+all-features = true
+rustdoc-args = ["--cfg", "docsrs"]
```

Comment:
do you know if we can have a single `[workspace.metadata.docs.rs]` in the workspace's `Cargo.toml` instead of in each package? i think `rustdoc` will behave similarly as `cargo`, so this might work.

Reply:
not sure 
but I think this is not possible because `docsrs` creates the documentation for each `crate` separately

Reply:
i check if I can build the workspace with `dorcsrs`

Reply:
> i check if I can build the workspace with dorcsrs

unfortunately doesn't seem to work

Reply:
i meant this differently: let's say we delete the `[package.metadata.docs.rs]` tables from the packages' `Cargo.toml`s and add a `[workspace.metadata.docs.rs]` table in the workspace's `Cargo.toml`. if we then call `cargo run ...` for `docs.rs` on a package, for example `xaynet-core` it should try to look up the metadata in the package's toml and since it doesn't find it, it should then look in the workspace's toml.

Reply:
ahh I see
I will check

Reply:
I checked but it does not seem to work (at least when building the docs locally)
I think `docsrs` downloads the crate from crates.io and therefore it does not have access to the workspace file 
but I could be wrong

Reply:
ok, then we have to keep the metadata in every package.